V6 rdm

Clarence-F opened the pull request:
RDM page (and Factory Reset as collateral) for v6 Help

Commit:
Web API documentation and project configuration

Change-Id: Ie579d516d2eab0ed4e9b341c0da8487f2b5ab26a

Commit:
Add instructions for Windows to readme

Change-Id: I754295ef2a2c6f53b01748b85c7f3ab42aef67e5

Commit:
Fix path in readthedocs configuration

Change-Id: I52d6c8f38711cf3b1cd3d370e8b9ad35522b3cfd

Clarence-F committed:
RDM added

Trial addition of adding RDM and Factory Reset to the YAML

Move Factory Reset to separate Branch

RDM Update

Update for the RDM Poll page

Update rdm.rst

Made some of Simon's changes

Update rdm.rst

Missed "params" call for start_rdm_discovery

Update rdm.rst

Fixed ToDos (again?)

sjmh3 commented on sphinx-src/snippets/pid-enums.rst:
```diff
@@ -0,0 +1,21 @@
+Current RDM implementation supports the following PIDs:
```
I think this is not useful as just a list of PIDs. They need to be shown in a table with the meta data required for a SET, if applicable, and the data returned for a GET.

sjmh3 commented on sphinx-src/web-api/factory-reset.rst:
```diff
@@ -0,0 +1,12 @@
+Factory Reset
+##############
```
```suggestion
#############
```

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+HTTP
+****
+
+.. _discover-http-get:
```
This is global, so should be e.g. rdm-discover-http-get

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+
+``GET /api/rdm/discovery``
+
+Returns a JSON object with the following attributes:
```
Returns an array - please see Crestron docs as shared.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+GET
+===
+
+``GET /api/rdm/discovery``
```
Needs a query param to specify the universe. Adam to update yaml.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+====
+
+``POST /api/rdm/discovery``
+Triggers an RDM poll of the defined universe
```
New line before this line for consistency

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+``POST /api/rdm/discovery``
+Triggers an RDM poll of the defined universe
+
+.. list-table::
```
What's this table showing? Needs an into.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+
+start_rdm_discovery
+===================
+``start_rdm_discovery(params, callback)``
```
New line before this one please.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+RDM PID Information
+###################
+
+.. _pid-http-get:
```
rdm-http-get-pid

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+====
+
+``POST /api/rdm/get``
+Queues an RDM poll that will return PID information. This is available by subscribing to the relevant websocket. 
```
```suggestion

Queues an RDM GET command that will return parameter information. This is available by subscribing to the relevant websocket. 
```

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+   * - ``meta``
+     - object
+     - Optional metadata for the PID, i.e. query params.
+     - **TODO**
```
As I commented on the pid-enums file, I think you need a full table of PIDs with the optional meta data.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+     - Optional metadata for the PID, i.e. query params.
+     - **TODO**
+
+.. _pid-http-set:
```
rdm-http-set-pid

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+.. _pid-http-set:
+
+``POST /api/rdm/set``
+Queues an RDM command that will alter specific PID information. This is available by subscribing to the relevant websocket. 
```
```suggestion

Queues an RDM command that will modify specific device configuration. The response can be received by subscribing to the relevant websocket. 
```

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+   * - ``meta``
+     - object
+     - Optional metadata for the PID, i.e. query params.
+     - **TODO**
```
Same comment as for GET - you need a table to show all meta data with the PIDs

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+
+start_rdm_get
+=============
+``start_rdm_get(params, callback)``
```
New line please

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+=============
+``start_rdm_get(params, callback)``
+
+Returns chosen PID information from the selected fixture.
```
```suggestion
Returns requested parameter information from the specified fixture.
```

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+   * - ``meta``
+     - object
+     - Optional metadata for the PID, i.e. query params.
+     - **TODO**
```
Same here - full table needed

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+
+start_rdm_set
+=============
+``start_rdm_set(params, callback)``
```
New line

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+   * - ``meta``
+     - object
+     - Optional metadata for the PID, i.e. query params.
+     - **TODO**
```
Full table

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+
+Universe Key String Format
+**************************
+.. _universe-key-string-format:
```
Goes above the heading

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+
+PID Enums
+*********
+.. _pid-enums:
```
Above the heading, and use new lines

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+RDM Discovery
+#############
+
+Discover RDM fixtures and get cached device info from discovered fixtures.
```
```suggestion
Discover RDM fixtures and get cached device info for discovered fixtures.
```

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+     - Value Example
+   * - ``uid``
+     - string
+     - manuId:deviceId[:subId] (Hex, 4byte:8byte[:decimal integer)
```
This could use a bit more description. You could use a snippet/reference this definition elsewhere.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+     - ``"072c:0004fe02"``
+   * - ``rdm_protocol_version``
+     - integer
+     - range of 0-65535
```
Descriptions are sentences - please start with a capital letter.

sjmh3 commented on sphinx-src/web-api/rdm.rst:
```diff
@@ -0,0 +1,259 @@
+     - range of 0-255
+     - ``0``
+
+.. _discover-http-post:
```
rdm-discover-http-post

Clarence-F committed:
Changes required partially complete

Update with WIP before handing back over to Simon

Clarence-F committed:
Update pid-enums.rst

Updated the PID enums to be into a table, but the data is incomplete from the twiki.
